recipes/libheif/all/patches/0001-cmake-1.14.1.patch

```diff
@@ -0,0 +1,75 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index d88a36a..5e6f116 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -42,8 +42,6 @@ set(CMAKE_CXX_EXTENSIONS OFF)
+ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+ set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
+ 
+-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+-
+ # Create the compile command database for clang by default
+ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+ 
+@@ -72,7 +70,7 @@ macro(plugin_option variableName packageName displayName displayType defaultPlug
+     option(WITH_${variableName} "Build ${displayName} ${displayType}" ON)
+     option(WITH_${variableName}_PLUGIN "Build ${displayName} as a plugin" ${defaultPlugin})
+     if (WITH_${variableName})
+-        find_package(${packageName})
++        find_package(${packageName} REQUIRED CONFIG)
+     endif ()
+ 
+     if (${variableName}_FOUND AND WITH_${variableName}_PLUGIN AND PLUGIN_LOADING_SUPPORTED_AND_ENABLED)
+@@ -90,10 +88,10 @@ macro(plugin_option variableName packageName displayName displayType defaultPlug
+ endmacro()
+ 
+ plugin_option(LIBDE265 LIBDE265 "libde265" "HEIC decoder" OFF)
+-plugin_option(X265 X265 "x265" "HEIC encoder" OFF)
+-plugin_option(DAV1D DAV1D "Dav1d" "AVIF decoder" OFF)
+-plugin_option(AOM_ENCODER AOM "aom" "AVIF encoder" OFF)
+-plugin_option(AOM_DECODER AOM "aom" "AVIF decoder" OFF)
++plugin_option(X265 libx265 "x265" "HEIC encoder" OFF)
++plugin_option(DAV1D dav1d "Dav1d" "AVIF decoder" OFF)
++plugin_option(AOM_ENCODER libaom-av1 "aom" "AVIF encoder" OFF)
++plugin_option(AOM_DECODER libaom-av1 "aom" "AVIF decoder" OFF)
+ plugin_option(SvtEnc SvtEnc "Svt-av1" "AVIF encoder" ON)
+ plugin_option(RAV1E RAV1E "Rav1e" "AVIF encoder" ON)
+ 
+@@ -112,6 +110,7 @@ if(IS_ABSOLUTE "${CMAKE_INSTALL_INCLUDEDIR}")
+ else()
+     set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+ endif()
++if(0)
+ if (LIBDE265_FOUND)
+     list(APPEND REQUIRES_PRIVATE "libde265")
+     set(have_libde265 yes)
+@@ -146,6 +145,7 @@ if (AOM_ENCODER_FOUND OR RAV1E_FOUND)
+ else()
+     set(have_avif_encoder no)
+ endif()
++endif()
+ list(JOIN REQUIRES_PRIVATE " " REQUIRES_PRIVATE)
+ set(VERSION ${PROJECT_VERSION})
+ 
+@@ -166,8 +166,6 @@ if(WITH_EXAMPLES)
+     add_subdirectory (examples)
+ endif()
+ add_subdirectory (libheif)
+-add_subdirectory (gdk-pixbuf)
+-add_subdirectory (gnome)
+ 
+ 
+ # --- packaging (source code)
+diff --git a/libheif/plugins/CMakeLists.txt b/libheif/plugins/CMakeLists.txt
+index 70dc1b4..75b2aa3 100644
+--- a/libheif/plugins/CMakeLists.txt
++++ b/libheif/plugins/CMakeLists.txt
+@@ -48,7 +48,7 @@ plugin_compilation(x265 X265 X265 X265)
+ 
+ set(LIBDE265_sources heif_decoder_libde265.cc heif_decoder_libde265.h)
+ set(LIBDE265_extra_plugin_sources ../heif_image.cc ../error.cc)
+-plugin_compilation(libde265 LIBDE265 LIBDE265 LIBDE265)
++plugin_compilation(libde265 libde265 LIBDE265 LIBDE265)
+ 
+ set(DAV1D_sources heif_decoder_dav1d.cc heif_decoder_dav1d.h)
+ set(DAV1D_extra_plugin_sources)
```

Comment:
I'm pretty sure that this patch is broken when libx265, libaom-av1 or dav1d are enabled.

Reply:
@SpaceIm 
Thanks for your comment
I am sorry that there is a problem with my easy patch.
I'll look into it a bit.